src/checkout-buttons/strategies/amazon-pay-v2/amazon-pay-v2-button-strategy.ts

```diff
@@ -64,34 +66,28 @@ export default class AmazonPayV2ButtonStrategy implements CheckoutButtonStrategy
                 checkoutLanguage,
                 ledgerCurrency,
                 checkoutSessionMethod,
-                region,
                 extractAmazonCheckoutSessionId,
             },
         } = paymentMethod;
 
-        if (!merchantId) {
-            throw new InvalidArgumentError();
+        if (!merchantId || !ledgerCurrency || !shopPath) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
-        const amazonButtonOptions = {
+        return {
             merchantId,
+            createCheckoutSession: {
+                url: `${shopPath}/remote-checkout/${paymentMethod.id}/payment-session`,
+                method: checkoutSessionMethod,
+                extractAmazonCheckoutSessionId,
+            },
             sandbox: !!testMode,
-            checkoutLanguage,
             ledgerCurrency,
-            region,
+            checkoutLanguage,
             productType: cart && cart.lineItems.physicalItems.length === 0 ?
```

Comment:
theres a new method available `getShippableLineItemsCount(cart) > 0` that you could use instead